daemon/api_sideload_n_try.go

```diff
@@ -285,9 +285,31 @@ func sideloadSnap(st *state.State, snapFile *uploadedSnap, flags sideloadFlags)
 		return nil, InternalError(err.Error())
 	}
 
+	// These two are filled only for components
+	var compInfo *snap.ComponentInfo
+	var snapInfo *snap.Info
+
 	sideInfo, apiErr := readSideInfo(st, snapFile.tmpPath, snapFile.filename, flags, deviceCtx.Model())
 	if apiErr != nil {
-		return nil, apiErr
+		// TODO we need to address the installation of local but
```

Comment:
it would be good if the line with TODO contained the word component, will make it easier to grep for these later